File: labs OOP/2.3/C#/TextContainer.cs
using System;
using System.Collections.Generic;

namespace TextProcessing
{
    public class TextContainer
    {
        private List<IText> texts;

        public TextContainer()
        {
            texts = new List<IText>();
        }

        public void AddString(string s)
        {
            texts.Add(new String(s));
        }

        public void DeleteString(int index)
        {
            if (index >= 0 && index < texts.Count)
            {
                texts.RemoveAt(index);
            }
        }

        public void ClearText()
        {
            texts.Clear();
        }

        public void UppercaseFirstLetters()
        {
            foreach (var text in texts)
            {
                text.UppercaseFirstLetters();
            }
        }

        public string GetKey(string s)
        {
            return new String(s).GetFirstChars();
        }

        public int GetCountOfLength(int len)
        {
            int count = 0;
            foreach (var text in texts)
            {
                if (text.GetValue().Length == len)
                {
                    count++;
                }
            }
            return count;
        }

        public IReadOnlyList<IText> GetTexts()
        {
            return texts.AsReadOnly();
        }
    }
}

File: labs OOP/2.3/C++/TextContainer.cpp
#include "TextContainer.h"
#include "String.h"

void TextContainer::AddString(const std::string& s) {
    texts.push_back(std::make_unique<String>(s));
}

void TextContainer::DeleteString(size_t index) {
    if (index < texts.size()) {
        texts.erase(texts.begin() + index);
    }
}

void TextContainer::ClearText() {
    texts.clear();
}

void TextContainer::UppercaseFirstLetters() {
    for (auto& text : texts) {
        text->UppercaseFirstLetters();
    }
}

std::string TextContainer::GetKey(const std::string& s) const {
    String temp(s);
    return temp.GetFirstChars();
}

size_t TextContainer::GetCountOfLength(size_t len) const {
    size_t count = 0;
    for (const auto& text : texts) {
        if (text->GetValue().length() == len) {
            ++count;
        }
    }
    return count;
}

const std::vector<std::unique_ptr<Text>>& TextContainer::GetTexts() const {
    return texts;
}